homeassistant/components/watchyourlan/__init__.py

```diff
@@ -0,0 +1,38 @@
+"""Support for WatchYourLAN."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_SSL, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, UPDATE_FREQUENCY
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WatchYourLAN from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Load the update interval from options if available
+    update_interval = entry.options.get("update_interval", UPDATE_FREQUENCY)
+
+    # Store the config entry data in hass.data for use by platforms
+    hass.data[DOMAIN][entry.entry_id] = {
+        "host": entry.data[CONF_HOST],
+        "port": entry.data[CONF_PORT],
+        "ssl": entry.data[CONF_SSL],
+        "url": entry.data["url"],
+        "update_interval": update_interval,  # Use the user-configured interval or default
+    }
```

Comment:
All this data is already available in the entry and thus is not relevant to store in hass.data